### ISSUE TITLE
How to generate low-light clean image in the step of Noisy Image Synthesis

### ISSUE THREAD

**Issue by ProNoobLi** [Vandermode/ELD#4]
According to the paper, synthetic low-light clean images are from long exposure and divided by low light factors. Is it an appropriate approach or is there any method supporting it?
![image](https://user-images.githubusercontent.com/33815430/87410284-afc97c00-c5f8-11ea-9aab-0ed2b1f6c01e.png)
When I check the distribution of one pair of raw images from short and long exposure separately(exposure ratio=250).
![image](https://user-images.githubusercontent.com/33815430/87409768-03879580-c5f8-11ea-8b20-493ab447528a.png)
The distribution of short raw lies on [500, 600] around; while that of long raw lies on [500, 10000]. 
Then, long exposure / 250 lies around [2, 40], which definitely doesn't fit for the low-light distribution. Thus, the sythetic noisy images generated by such inaccurate low-light clean + generated noise doesn't make sense.

Is my understanding correct? Please point out the wrong things I made.

**Comment by Vandermode** [comment]
> The distribution of short raw lies on [500, 600] around

How many images you used?

**Comment by Vandermode** [comment]
> According to the paper, synthetic low-light clean images are from long exposure and divided by low light factors. Is it an appropriate approach or is there any method supporting it?

Yes, it's indeed a valid assumption. If there is no noise, the recorded image intensity increases linearly during exposure in raw space as raw is radiometrically calibrated (See [1] for details). That's one of the reason to model noise in raw space.

[1]   S. J. Kim, H. T. Lin, Z. Lu, S. Süsstrunk, S. Lin, and M. S. Brown, “A new in-camera imaging model for color computer vision and its application,” IEEE Transactions on Pattern Analysis and Machine Intelligence, vol. 34, no. 12, pp. 2289–2302, 2012.

To help you understand more things about raw space (and ISP), I would recommend the [excellent tutorial](https://www.eecs.yorku.ca/~mbrown/ICCV2019_Brown.html) by Michael Brown

**Comment by ProNoobLi** [comment]
Hi, thanks for your reply. 
I just check one image to verify that assumption.
However, could you please explain why the long_range / ratio doesn't match the short range? I mean it is far from the distribution.
Should I do long_raw / ratio directly or (long_raw - black_level) / ratio?

**Comment by ProNoobLi** [comment]
Thanks for your recommendation of the tutorial. I just want to clarify if my understanding is correct
![image](https://user-images.githubusercontent.com/33815430/87498104-a0464380-c689-11ea-87a1-45d7dcf0d968.png)
![image](https://user-images.githubusercontent.com/33815430/87498186-c370f300-c689-11ea-9df1-b4a215523375.png)
![image](https://user-images.githubusercontent.com/33815430/87498284-06cb6180-c68a-11ea-89a2-01122862b560.png)
For the first image, the intensities of images are nonlinear to the irradiance ex, and the nonlinear function f which including some unknown tone-mapping. However, the RAW is proportional to the irradiance ex and the number of collected photons thus could be linear to the exposure ratio k as well. The key points here is that RAW pixel isn't the intensity.
Is my understanding correct??

**Comment by Vandermode** [comment]
> For the first image, the intensities of images are nonlinear to the irradiance ex, and the nonlinear function f which including some unknown tone-mapping. However, the RAW is proportional to the irradiance ex and the number of collected photons thus could be linear to the exposure ratio k as well. The key points here is that RAW pixel isn't the intensity.
Is my understanding correct??

Yes, you are right.

Besides, before 2014, there were few cameras providing the raw image accessibility for users. So the researchers had to work on sRGB space. This leads to the study on radiometric calibration, with the goal to recover raw response that  is proportional to the scene irradiance

**Comment by ProNoobLi** [comment]
To calibrate the K in your paper, should I refer to this part, 3.2 Evaluation of constant K in [32]??
![image](https://user-images.githubusercontent.com/33815430/87746102-52187800-c822-11ea-8260-3ffd8b6833d7.png)
I notice that the K in your paper is with the unit DU/electrons; while the K from [32] is at electrons/DU. Right?

**Comment by Vandermode** [comment]
Yes, please. But, as a reminder, I'm unable to provide more implementation details of our methods beyond the paper

**Comment by XinYu-Andy** [comment]
Hello?I have the same question. The raw data mostly is 14-bit(16383). So if you use an factor in[100,300], then the maximum of the simulated low_light raw should be about [55, 163]?

**Comment by ProNoobLi** [comment]
> simulated low_light raw should be about [55, 163]?

Yup.
From my experience, the datasets I processed are smartphone pictures in 10bits which ranges from 0 to 1023. The raw values of some real extreme dark images might be around 65(while the black level is 64).

**Comment by fm123a** [comment]
hi,  i have a  question, the  low-light clean images are from long exposure and divided by low light factors, why  the  sythetic noisy images need Multiplied by the low light factors,   when the low-light clean images add the  generated noise , i think this step maybe change noise distribution.
